litex/soc/integration/soc_core.py

```diff
@@ -428,6 +429,9 @@ def add_csr_master(self, csrm):
             raise FinalizeError
         self._csr_masters.append(csrm)
 
+    def add_linker_region(self, name, origin, length):
+        self._linker_regions.append((name, origin, length))
```

Comment:
This should probably verify it the new `linker region` is included in any defined `memory region`.